src/Helpers/JWKFetcher.php

```diff
@@ -59,101 +62,135 @@ protected function convertCertToPem($cert)
         return $output;
     }
 
+    // phpcs:disable
     /**
-     * Fetch x509 cert for RS256 token decoding.
+     * TODO: Deprecate, use $this->getJwksX5c() instead.
+     *
+     * @param string $iss
+     *
+     * @return array|mixed|null
      *
-     * @param string      $domain Base domain for the JWKS, including scheme.
-     * @param string|null $kid    Kid to use.
-     * @param string      $path   Path to the JWKS from the $domain.
+     * @throws \Exception
      *
-     * @return mixed
      *
-     * @throws \Exception If the Guzzle HTTP client cannot complete the request.
+     * @codeCoverageIgnore
      */
-    public function fetchKeys($domain, $kid = null, $path = '.well-known/jwks.json')
+    public function fetchKeys($iss)
```

Comment:
FINE. This method is now deprecated. But why keep that logic? Is mostly what you are doing now on the new one, right? Let's remove this body and call the new method passing the right parameters and making sure what we return sticks to the "legacy" format this now-deprecated-method was originally returning. I think it would be something like:

```php 
//this was receiving just the "base url" without the jwks path
$url = "{$iss}.well-known/jwks.json";
//call the new method ;)
jwk = getJwksX5c($url);
//return an array with the single key or empty array if missing
return [jwk];
```

Reply:
Different return value makes this a bad candidate for refactoring using the new method.